feature/google/src/bc5SupportDev/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -0,0 +1,731 @@
+                        onError(purchasesError)
+                        return@queryInAppsPurchasesAsync
+                    }
+                    val mapOfUnconsumedInApps =
+                        unconsumedInAppsPurchases.toMapOfGooglePurchaseWrapper(BillingClient.ProductType.INAPP)
+                    onSuccess(mapOfActiveSubscriptions + mapOfUnconsumedInApps)
+                }
+            }
+        }
+    }
+
+    private fun List<Purchase>.toMapOfGooglePurchaseWrapper(
+        @BillingClient.ProductType productType: String
+    ): Map<String, StoreTransaction> {
+        return this.associate { purchase ->
+            val hash = purchase.purchaseToken.sha1()
+            hash to purchase.toStoreTransaction(productType.toProductType(), presentedOfferingIdentifier = null)
+        }
+    }
+
+    override fun findPurchaseInPurchaseHistory(
+        appUserID: String,
+        productType: ProductType,
+        sku: String,
+        onCompletion: (StoreTransaction) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        withConnectedClient {
+            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE_WITH_TYPE.format(sku, productType.name))
+            productType.toGoogleProductType()?.let { googleProductType ->
+                queryPurchaseHistoryAsyncEnsuringOneResponse(googleProductType) { result, purchasesList ->
+                    if (result.isSuccessful()) {
+                        val purchaseHistoryRecordWrapper =
+                            purchasesList?.firstOrNull { it.listOfSkus.contains(sku) }
+                                ?.toStoreTransaction(productType)
+
+                        if (purchaseHistoryRecordWrapper != null) {
+                            onCompletion(purchaseHistoryRecordWrapper)
+                        } else {
+                            val message = PurchaseStrings.NO_EXISTING_PURCHASE.format(sku)
+                            val error = PurchasesError(PurchasesErrorCode.PurchaseInvalidError, message)
+                            onError(error)
+                        }
+                    } else {
+                        val underlyingErrorMessage = PurchaseStrings.ERROR_FINDING_PURCHASE.format(sku)
+                        val error =
+                            result.responseCode.billingResponseToPurchasesError(underlyingErrorMessage)
+                        onError(error)
+                    }
+                }
+            } ?: onError(
+                PurchasesError(PurchasesErrorCode.PurchaseInvalidError, PurchaseStrings.NOT_RECOGNIZED_PRODUCT_TYPE)
+            )
+        }
+    }
+
+    @VisibleForTesting(otherwise = PRIVATE)
+    @Suppress("ReturnCount")
+    internal fun getPurchaseType(purchaseToken: String, listener: (ProductType) -> Unit) {
+        billingClient?.let { client ->
+            val querySubsPurchasesParams =
+                QueryPurchasesParams.newBuilder().setProductType(BillingClient.ProductType.SUBS).build()
```

Comment:
I just had a random idea, but this could be an extension function on `BillingClient.ProductType`. Something like `BillingClient.ProductType.buildQueryPurchasesParams()`

Not needed but just in case you like it

Reply:
yessss i actually had that on my list and forgot it, thanks for reminding me!